mod/primitives/pkg/transition/validator_update.go

```diff
@@ -33,3 +38,26 @@ type ValidatorUpdate struct {
 	// EffectiveBalance is the effective balance of the validator.
 	EffectiveBalance math.Gwei
 }
+
+// RemoveDuplicates removes duplicate validator updates.
+func (vu ValidatorUpdates) RemoveDuplicates() ValidatorUpdates {
```

Comment:
@ocnc can you review, you did original work here

Reply:
I see what you're doing but I think in the original logic, we were taking the latest occurrence of a val update (whatever was appended last to the val updates array is the update we want to retain), and this logic takes the first occurrence instead

Reply:
Because of how the process multiple slots works => slot1, slot2, slot3, and the updates from slot3 should override slot1

Reply:
addressed